native/src/main/java/tech/sourced/babelfish/JsonASTVisitor.java

```diff
@@ -933,11 +1017,10 @@ public int visit(IASTDeclarator node) {
                     ICPPASTDeclarator impl = (ICPPASTDeclarator) node;
                     json.writeBooleanField("DeclaresParameterPack", impl.declaresParameterPack());
 
-                    if (lastTypeVisited != null) {
+                    if (!typesVisited.empty()) {
```

Comment:
Should we be reparenting _all_ the types in the collection?  Or is this just the stack and we're reparenting the top?

(But if it's the latter: Do we want to _pop_ the stack, or just modify it in place?)

Reply:
From the comment I added by your suggestion above:

```c
        // These are type nodes for parameters. CDT gives them as siblings of the
        // parameter node; we reparent them as a child of the parameter, so we store
        // them in a stack and pop them on the next parameter node.
```

So a Parameters field will be in CDT a list like: `[TypeNode, ParameterNode, TypeNode, ParameterNode, TypeNode, Parameter]`. What we do is store TypeNodes that are inside parameters lists and put them as children of the next parameter. But what I didn't account for before, and thus caused a bug, is that sometimes the `ParameterNode` can be some kind of node with a `Parameter` list itself, for example a template definition or a function pointer (or both, like the ones used for `std::function` for example) so the fix is to use a stack so the latest added types are the first to be removed.